### PR TITLE
MT-751: Migrate to Typescript & Function Components

### PR DESCRIPTION
**Pull request by DidierF**

### CODE REVIEW

File: src/BlockedUsers.tsx [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        appColor={appColor}
+        onLayout={({ nativeEvent: ne }: { nativeEvent: any }) =>
+          (itemHeight.current = ne.layout.height)
+        }
+        item={{ user: item }}
+        center
+        admin={admin}
+        type={'banned'}
+        onTap={() => floatingMenu.current?.close()}
+        onToggleBlockStudent={() => {
```
Comment (csillaj) [inline src/BlockedUsers.tsx]: please don't use inline functions

File: src/BlockedUsers.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        </TouchableOpacity>
+      </View>
+      {loading ? (
+        <ActivityIndicator
+          size='large'
+          color={isDark ? 'white' : 'black'}
+          style={styles.activityIndicator}
+        />
+      ) : (
+        <>
```
Comment (csillaj) [inline src/BlockedUsers.tsx]: Is this extra needed?
`(
<>
....
</>
)
`

File: src/BlockedUsers.tsx (hunk 1 above)
Comment (csillaj) [reply src/BlockedUsers.tsx]: and remove inline styles where possible

File: src/ChatList.tsx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+} from 'react-native';
+import { arrowDown, sendMsg, x } from './svgs';
+import { FormatMessageResponse, UserResponse } from 'stream-chat';
+import ListItem from './ListItem';
+import { IChatType } from './types';
+
+interface IChatList {
+  appColor: string;
+  isDark: boolean;
+  isiOS: boolean;
```
Comment (csillaj) [inline src/ChatList.tsx]: is it worth passing isIos as a prop?

File: package.json [hunk 4]
```diff
@@ -5,9 +5,27 @@
   "peerDependencies": {
     "react": "18.2.0",
     "react-native": "0.71.4",
-    "stream-chat": "^3.7.0",
     "react-native-android-keyboard-adjust": "^1.2.0",
+    "react-native-device-info": "^10.9.0",
+    "react-native-safe-area-context": "^3.4.1",
     "RNDownload": "git+https://github.com/railroadmedia/RNDownload.git#master-musora-app",
-    "react-native-device-info": "^10.9.0"
+    "stream-chat": "^3.7.0"
```
Comment (csillaj) [inline package.json]: might be worth updating to latest version, I looked into the methods and I don't think they changed syntax or method names

**Comment by csillaj** [comment]
I have some type errors in MusoraChat.tsx, mostly "Parameter xyz implicitly has an 'any' type"

File: src/ChatList.tsx (hunk 3 above)
Comment (DidierF) [reply src/ChatList.tsx]: Removed it.

File: src/BlockedUsers.tsx (hunk 2 above)
Comment (DidierF) [reply src/BlockedUsers.tsx]: Nope, removed it.

File: src/BlockedUsers.tsx (hunk 1 above)
Comment (DidierF) [reply src/BlockedUsers.tsx]: Done!

**Comment by DidierF** [comment]
@csillaj I got those errors after updating to the latest version, will check them.

**Review by MeenuDhanoa — APPROVED** [review]

File: src/MusoraChat.tsx [hunk 5]
```diff
@@ -0,0 +1,677 @@
...
+          me={me}
+          pinned={pinned}
+          messages={messages}
+          hidden={hidden}
+          client={client}
+          onMessageTap={() => floatingMenu.current?.close?.()}
+          handleMessage={handleMessage}
+          onScroll={onChatScroll}
+          loadMore={loadMore}
+          onTextBoxPress={() => {
```
Comment (csillaj) [inline src/MusoraChat.tsx]: remove inline

File: src/MusoraChat.tsx [hunk 6]
```diff
@@ -0,0 +1,677 @@
...
+        />
+      ) : showParticipants ? (
+        <Participants
+          isDark={isDark}
+          appColor={appColor}
+          admin={me?.role === 'admin'}
+          onlineUsers={tabIndex ? questionsViewers : chatViewers}
+          channel={currentChannel}
+          onBack={() => setShowParticipants(false)}
+          onBlockedStudents={() => {
```
Comment (csillaj) [inline src/MusoraChat.tsx]: remove inline

File: src/MusoraChat.tsx [hunk 7]
```diff
@@ -0,0 +1,677 @@
...
+          }}
+        />
+      ) : showBlocked ? (
+        <BlockedUsers
+          isDark={isDark}
+          appColor={appColor}
+          admin={me?.role === 'admin'}
+          client={client}
+          onBack={() => setShowBlocked(false)}
+          onParticipants={() => {
```
Comment (csillaj) [inline src/MusoraChat.tsx]: remove inline

File: src/MusoraChat.tsx [hunk 8]
```diff
@@ -0,0 +1,677 @@
...
+        />
+      ) : (
+        <>
+          <TabMenu
+            isDark={isDark}
+            appColor={appColor}
+            tabIndex={tabIndex}
+            isLandscape={isLandscape}
+            showResources={!!resources.length}
+            onTabChange={i => {
```
Comment (csillaj) [inline src/MusoraChat.tsx]: remove inline

File: src/MusoraChat.tsx [hunk 9]
```diff
@@ -0,0 +1,677 @@
...
+            onBlockedStudents={() => setShowBlocked(true)}
+          />
+          <TextBoxModal
+            onClose={() => setKeyboardVisible(false)}
+            onShow={() => setTimeout(() => commentTextInput.current?.focus(), isiOS ? 0 : 100)}
+            visible={keyboardVisible}
+            isDark={isDark}
+            ref={commentTextInput}
+            onSubmitEditing={handleMessage}
+            onChangeText={changedComment => {
```
Comment (csillaj) [inline src/MusoraChat.tsx]: remove inline

**Review by csillaj — APPROVED** [review]

**Comment by csillaj** [comment]
@DidierF shouldn't this be tested before merging it in ?

**Comment by DidierF** [comment]
@csillaj I didn't merge the app branch, the ticket is in QA.

**Pull request merged**